li-apache-kafka-clients/src/main/java/com/linkedin/kafka/clients/consumer/FederatedSubscriptionState.java

```diff
@@ -0,0 +1,90 @@
+/*
+ * Copyright 2019 LinkedIn Corp. Licensed under the BSD 2-Clause License (the "License").  See License in the project root for license information.
+ */
+
+package com.linkedin.kafka.clients.consumer;
+
+import java.util.Collections;
+import java.util.HashSet;
+import java.util.Set;
+import java.util.regex.Pattern;
+
+import org.apache.kafka.common.TopicPartition;
+
+// Current subscription/assignment state at the federated level
+public class FederatedSubscriptionState {
+  public enum SubscriptionType {
+    NONE, MANUAL_ASSIGNMENT, SUBSCRIPTION, PATTERN_SUBSCRIPTION
+  }
+
+  private SubscriptionType _subscriptionType;
```

Comment:
no need to have the field - just hardcode the getter in the subclasses

Reply:
done